### ISSUE TITLE
QuotaExceededError using Ionic Storage with a few strings

### ISSUE THREAD

**Issue by ManigandanRaamanathan** [localForage/localForage#828]
Ionic Storage Issue for less memory space
I'm having this issue with Ionic Storage, this is the complete stack trace:

> core.es5.js:1084 ERROR Error: Uncaught (in promise): QuotaExceededError: 
> at c (polyfills.js:3)
> at c (polyfills.js:3)
> at polyfills.js:3
> at t.invokeTask (polyfills.js:3)
> at Object.onInvokeTask (core.es5.js:4140)
> at t.invokeTask (polyfills.js:3)
> at r.runTask (polyfills.js:3)
> at o (polyfills.js:3)
> at IDBTransaction.invoke (polyfills.js:3)

This happen both in mobile and computer  with low internal storage. If the device storage is below 1GB then this storage module freezes entirely and wont respond until freeing space more than 1GB. We need immediate attention and fix for this problem.

**Comment by tofumatt** [comment]
You're using Ionic, but for which platform? Having only 1GB of storage left, especially on a PC, doesn't surprise me that a quota error is encountered. What kind of mobile device are you encountering the error on? What kind of data are you attempting to save when you encounter the error?

This sounds like it's working as intended 🤷‍♂️

**Comment by ManigandanRaamanathan** [comment]
Facing this issue on android devices. Tried Samsung J7 and note 5 with 900MB device internal storage. Both of them not working if the device storage is below 1GB. After clearing space out >=1GB then only it works

**Comment by ManigandanRaamanathan** [comment]
Any update on how to fix this issue?

**Comment by thgreasi** [comment]
First of all let's clarify which storage driver is used. Since you mentioned J7 I would guess that would be IndexedDB (except if you pack your Ionic app as an apk?)

**Comment by ManigandanRaamanathan** [comment]
How to identify the storage that the app using in the actual device?

**Comment by thgreasi** [comment]
You can run `localforage.driver()`, ideally inside the error handler of the failed storage method.
Please also clarify whether the error originates from a browser instance of your app or if you are providing it as a store app.

**Comment by ManigandanRaamanathan** [comment]
The below screenshot is taken from chrome inspect tool while debugging in my Samsung j7 android device which is currently having 800MB. It shows the storage driver is " asyncStorage" and shows that error. If I clear up some space around more than 1GB then it will work fine. It never really use sqlite or other drivers here. Is that could be the issue?

![capture](https://user-images.githubusercontent.com/22089729/41767990-a8f2efb8-763d-11e8-871e-352732f65e9c.png)

**Comment by thgreasi** [comment]
If you are packaging you ionic app as an apk, then you can use the [cordovaSQLiteDriver](https://github.com/thgreasi/localForage-cordovaSQLiteDriver), which should give more disk space to your app. If you are just providing a web app, then cordiva is obviously not an option.

**Comment by ManigandanRaamanathan** [comment]
Is there any other alternative solution to fix within the existing storage plugin?

**Comment by ManigandanRaamanathan** [comment]
Wait. Did you mean i have to download and add ionic cordova sqlite on top of the default storage for the sqlite to kick in when deploying apk in the actual device? Is that correct?

**Comment by thgreasi** [comment]
Yes that's what I meant but you never answered on whether you are packing an apk or serve as a web app. I thought that ionic storage should add use the Cordova sqlite plugging out of the box. How do you call the storage methods in your code? (Eg `localforage.setItem()`)
